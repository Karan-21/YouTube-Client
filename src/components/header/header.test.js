// ComponentName.test.js
import React from 'react';
import { configure } from 'enzyme'

import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Header from './Header';
import { BrowserRouter } from 'react-router-dom';
configure({adapter: new Adapter()});

// Inside describe we can add nested blocks for different tests
// First Test case checks where we are Header component is rendered correctly by taking SnapShot.
describe("Header Compoment", () => {
  it("should render Header", () => {
    const wrapper = shallow(<Header />)
    expect(wrapper.getElements()).toMatchSnapshot();

  });

  // Second Test case checks where inital App is loaded with Blank Input fields.
  it("App loads with initial state of empty", () => {
    const wrapper = shallow(<Header />);
    const text = wrapper.find("#input-search").text();
    expect(text).toEqual("");
  });

  // Third Test case check if the user is typing inside the Input field then the Correct value is added to the UI and Database.
  it('should change the input value', () => {
    let wrapper = shallow(<Header/>);
    const input = wrapper.find("#input-search")
    input.simulate('change', { target: { value: 'Hello' } })
    expect(wrapper.find("#input-search").get(0).props.value).toEqual('Hello');
  })

});