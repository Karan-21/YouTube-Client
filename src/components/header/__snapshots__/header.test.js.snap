// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentName should render my component 1`] = `
Array [
  <div
    className="header"
  >
    <div
      className="header__left"
    >
      <Memo />
      <Link
        to="/"
      >
        <img
          alt="yt_logo"
          className="header__logo"
          src="logo.png"
        />
      </Link>
    </div>
    <form
      className="header__input"
      onSubmit={[Function]}
    >
      <input
        id="input-search"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
      <Link
        to="/search/"
      >
        <Memo
          className="header__inputButton"
        />
      </Link>
    </form>
    <div
      className="header__icons"
    >
      <React.Fragment>
        <Memo
          className="header__icon"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        />
        <Memo
          className="header__icon"
        />
        <Memo
          className="header__icon"
        />
        <WithStyles(ForwardRef(Avatar))
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        />
        <SimpleDialog
          onClose={[Function]}
          open={false}
        />
        <WithStyles(ForwardRef(Menu))
          anchorEl={null}
          id="simple-menu"
          keepMounted={true}
          onClose={[Function]}
          open={false}
        >
          <WithStyles(ForwardRef(MenuItem))
            onClick={[Function]}
          >
            Sign In
          </WithStyles(ForwardRef(MenuItem))>
        </WithStyles(ForwardRef(Menu))>
      </React.Fragment>
    </div>
  </div>,
]
`;

exports[`Header Compoment should render Header 1`] = `
Array [
  <div
    className="header"
  >
    <div
      className="header__left"
    >
      <Memo />
      <Link
        to="/"
      >
        <img
          alt="yt_logo"
          className="header__logo"
          src="logo.png"
        />
      </Link>
    </div>
    <form
      className="header__input"
      onSubmit={[Function]}
    >
      <input
        id="input-search"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
      <Link
        to="/search/"
      >
        <Memo
          className="header__inputButton"
        />
      </Link>
    </form>
    <div
      className="header__icons"
    >
      <React.Fragment>
        <Memo
          className="header__icon"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        />
        <Memo
          className="header__icon"
        />
        <Memo
          className="header__icon"
        />
        <WithStyles(ForwardRef(Avatar))
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        />
        <SimpleDialog
          onClose={[Function]}
          open={false}
        />
        <WithStyles(ForwardRef(Menu))
          anchorEl={null}
          id="simple-menu"
          keepMounted={true}
          onClose={[Function]}
          open={false}
        >
          <WithStyles(ForwardRef(MenuItem))
            onClick={[Function]}
          >
            Sign In
          </WithStyles(ForwardRef(MenuItem))>
        </WithStyles(ForwardRef(Menu))>
      </React.Fragment>
    </div>
  </div>,
]
`;
